### PR TITLE
CDAP-13872 add mmds docs

### PR DESCRIPTION
**Pull request by albertshau**

### CODE REVIEW

**Comment by albertshau** [comment]
https://builds.cask.co/browse/CDAP-DQB409

**Comment by albertshau** [comment]
https://builds.cask.co/artifact/CDAP-DQB409/shared/build-1/Docs-HTML/html/user-guide/mmds/index.html

File: cdap-docs/_common/_source/guides.rst [hunk 1]
```diff
@@ -34,6 +37,8 @@ Guides
 
   - |user-guide-pipe-black|_ Documentation on usage of Data Pipelines UI
 
+  - |user-guide-mmds-black|_ Interactive, UI driven analytics and machine learning
```
Comment (anwar6953) [inline cdap-docs/_common/_source/guides.rst]: should it be `UI-driven`?

File: cdap-docs/user-guide/source/mmds/concepts.rst [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+In addition to a data source, each experiment specifies an outcome. The outcome is the field in your
+data that you want to be able to predict. An outcome is either a categorical outcome, or a numeric outcome.
+Categorical outcomes are discrete values, where each value represents a different category.
+Numerical outcomes are continuous numbers that represent data on a sliding scale. For example, if you are trying to
+predict if a new hire will leave the company within a year, your outcome is categorical -- true or false.
+If you are trying to predict that sale price of a house on the market, your outcome is numeric.
+
+Each experiment contains zero or more models. Since the experiment outcome is fixed, all models in an
+experiment will be trained to predict the same thing. An experiment is a way to group related models
+together in one place and compare how each of them perform. In the realestate example, all models in the
```
Comment (anwar6953) [inline cdap-docs/user-guide/source/mmds/concepts.rst]: Should there be a space between "real" and "estate"?
Or are you treating "realestate" as a name of the example?

File: cdap-docs/user-guide/source/mmds/concepts.rst [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+Different algorithms are available depending on the outcome type. If the outcome type is categorical,
+classification algorithms can be used to train the model. If the outcome type is numeric,
+regression algorithms can be used. Each algorithm has a set of parameters that can be adjusted to try
+and produce a better model. Many times, trial and error is used to figure out a good set of parameters.
+
+Evaluation Metrics
+^^^^^^^^^^^^^^^^^^
+
+During the model training process, the input data is split into a training dataset and a test dataset. By default,
+the training dataset is 80% of the input data and is randomnly chosen. Any data point not in the training dataset
```
Comment (anwar6953) [inline cdap-docs/user-guide/source/mmds/concepts.rst]: nit - s/randomnly/randomly

File: cdap-docs/user-guide/source/mmds/concepts.rst [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+is not spam.
+
+This model will have good precision because it will correctly predict an email to be non spam 90% of the time. However,
+it has poor recall because it will never correctly label spam emails.
+Precision usually increases at the expense of recall and recall usually increases at the expense of precision.
+People often optimize for f1 as a way to find a happy balance between precision and recall.
+
+For numeric outcomes, root mean squared error (RMSE), mean absolute error, R2, and explained variance are used. Lower values are better.
+
+Mean absolute error is the average of the error between the prediction and the observed outcome
```
Comment (anwar6953) [inline cdap-docs/user-guide/source/mmds/concepts.rst]: nit - how come no periods at the end of these definitions, unlike for lines 86-90?

File: cdap-docs/user-guide/source/mmds/concepts.rst [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+Explained variance is measure related to R2 of how much of the error between prediction and observed outcome is explained by the model
+
+
+Predictions Dataset
+^^^^^^^^^^^^^^^^^^^
+
+By default, 20% of the input data source will be used to evaluate the model. The predictions generated during model evaluation are saved
+so that you can examine the them after model training to try and figure out how to train a better model. The predictions dataset can be
+explored using SQL to get a better understanding of how the model behaves, and possible feature cleaning that could be done to improve
+the model. For example, suppose you notice that house price predictions tend to farther off when the number of bedrooms is not known.
```
Comment (anwar6953) [inline cdap-docs/user-guide/source/mmds/concepts.rst]: "to farther off" --> "to be farther off"

File: cdap-docs/user-guide/source/mmds/example.rst [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+This takes us back to the Data Preparation view.
+
+.. figure:: /_images/mmds-example-filter-price.png
+  :figwidth: 100%
+  :width: 800px
+  :align: center
+  :class: bordered-image
+
+Add a filter on the price column such that we only keep rows that have a price between one and ten million
+dollars. Houses in the Bay Area are expensive.
```
Comment (anwar6953) [inline cdap-docs/user-guide/source/mmds/example.rst]: :(

File: cdap-docs/user-guide/source/mmds/example.rst [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+  :width: 800px
+  :align: center
+  :class: bordered-image
+
+For example, we can run a SQL query to see which cities have the worst predictions.
+Normally at this point, you would explore the predictions some more or try training
+other models with different modeling algorithms and parameters to try and generate
+a model with the least amount of error. Once you have created a model that you are
+happy with, you can create a scoring pipeline with the model. This will bring you
+to the Pipeline Studio with part of a pipeline pre-configured for you. All that is left
```
Comment (anwar6953) [inline cdap-docs/user-guide/source/mmds/example.rst]: nit - preconfigured (without the dash?)

File: cdap-docs/user-guide/source/mmds/feature-gen.rst [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+below, we have parsed our input, renamed columns, set various data types, and filtered out
+real estate prices that are under one million dollars.
+
+.. figure:: /_images/mmds-feature-gen.png
+  :figwidth: 100%
+  :width: 800px
+  :align: center
+  :class: bordered-image
+
+In this example, the outcome is price, and the features are city, zip, type, beds, baths, sqft, lot, stories, and year_built.
```
Comment (anwar6953) [inline cdap-docs/user-guide/source/mmds/feature-gen.rst]: `lot` --> `lot_sqft`?
You reference the column name as `lot_sqft` in https://github.com/caskdata/cdap/pull/10443/files#diff-f272ee6a7f9e505893c993c6dc040904R42

**Review by anwar6953 — CHANGES_REQUESTED** [review]
Only had minor comments.
Nice work on having an easy to read and understand guide!

**Comment by albertshau** [comment]
docs for latest commit, with more example detail https://builds.cask.co/artifact/CDAP-DQB409/shared/build-2/Docs-HTML/html/user-guide/mmds/example.html

**Comment by deb-saunders** [comment]
Docs reviewed and changes requested as per conversation with albertshau@

**Comment by albertshau** [comment]
docs after latest build https://builds.cask.co/artifact/CDAP-DQB409/shared/build-3/Docs-HTML/html/user-guide/mmds/index.html

**Pull request merged**